New docs website / Update component api sections

alex-ju opened the pull request:
### :pushpin: Summary

Update `Doc::ComponentApi` to match design.

### :hammer_and_wrench: Detailed description

We update the `@value` argument to take an array instead of a comma-separated string (aligning it with the `Doc::WcagList` component and the HTML structure and CSS to match the new design.

[👉 Preview link](https://hds-website-749oxj13j-hashicorp.vercel.app/components/alert/)

### :camera_flash: Screenshots

<table>
<tr><th>Before</th><th>After</th></tr>
<tr><td valign="top">

![component-api-before](https://user-images.githubusercontent.com/788096/207033056-70e6936b-cd44-40af-932d-8500144c178f.png)

</td><td>

![component-api-after](https://user-images.githubusercontent.com/788096/207033072-3a0b82d6-382c-449c-a904-104d17c9a39c.png)


</td></tr>
</table>


### :link: External links

[Jira issue](https://hashicorp.atlassian.net/browse/HDS-1056)
[Design reference](https://www.figma.com/file/Ky0qWjvHZR3je1lCBlzNB1/HDS-website?node-id=1364%3A85506&t=kdnlBGICNMkMM1GC-1)

***

### 👀 How to review

<!-- Suggest how it's best for the reviewer to review the code (choose one, or remove) -->
👉 Review commit-by-commit

Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

alex-ju commented on website/app/styles/components/component-api.scss:
```diff
@@ -0,0 +1,160 @@
+  font-size: 0.75rem;
+  line-height: 1.5;
+
+  &--name {
```
I generally don't like chaining modifiers (for 'searchability' reasons, mainly) but in this case it would save us from quite a bit of code repetition

didoo commented on website/app/styles/components/component-api.scss:
```diff
@@ -0,0 +1,160 @@
+  grid-template-columns: repeat(3, 1fr);
+  margin: 0;
+  padding: 24px 16px;
+  border-bottom: 1px solid #ccc;
```
if this is not a standard color (token) on purpose, by design, please add a comment so we know it's intentional

didoo commented on website/app/styles/components/component-api.scss:
```diff
@@ -0,0 +1,160 @@
+}
+
+.doc-component-api__term {
+  @include doc-font-family("gilmer");
```
check if the new typography mixins can help you here

alex-ju replied:
yes `doc-font-style-label`

didoo reviewed (approved):
All good on the code side. 👍

alex-ju commented:
@didoo thanks for the review – I rebased to `main` and added 2 more commits (kept them separately to ease the review); let me know what you think, especially about the placement of `sr-only` utility class (ported from the dummy app).

didoo commented on website/app/styles/global/index.scss:
```diff
@@ -1,3 +1,4 @@
 // GLOBAL STYLES
 
 @import "./link";
+@import "./sr-only";
```
probably for consistency with what we're doing in the HDS components, I would move this to the `website/app/styles/app.scss` file (it's a global declaration, and is special)

alex-ju replied:
cool, moved it to `app.scss`

didoo commented on website/app/styles/global/sr-only.scss:
```diff
@@ -0,0 +1,12 @@
+.doc-sr-only {
```
nice catch! didn't realized I was using the HDS components one! :)

heatherlarsen reviewed (approved):
@alex-ju This is a thing of beauty! 😍 Thank you for your thoughtful approach to this complex component.

alex-ju commented on website/app/styles/components/component-api.scss:
```diff
@@ -0,0 +1,157 @@
+  color: var(--doc-color-gray-300);
+  font-weight: 700;
+  font-size: 0.875rem;
+  line-height: 1.357;
```
Good call – I added an "outlined" type to the `Doc::Badge` component and removed these bespoke styles.
@didoo if I have your thumbs-up on this late addition I'll get it in.
